Feature/circular_stats

k-fleck-01 opened the pull request:
Added circular statistics (mean, variance and standard deviation) implementation to "src/output/stats.rs". Added an updated example, "stats.yml", that includes both circular statistics and the expressions subkey - would replace the "stats_expr.yml" from the stats-expr PR ideally.

Checked that circular results agree with linear statistics in the small angle approximation too.

k-fleck-01 committed:
Implemented circular statistics

k-fleck-01 committed:
Added example and comments.

tgblackburn commented:
Thanks Kyle, looks good. Again, a brief explanation of the new stats functions in input.md would be great.

k-fleck-01 committed:
Fixed PR review comments

k-fleck-01 commented:
Added the extra choices "circvariance" and "cvariance" for the circular reduction and added some info to the input.md about the new circular stats options.

tgblackburn committed:
Merge branch 'devel' into feature/circular_stats

k-fleck-01 commented on src/output/stats.rs:
I can try implementing an angle check, not sure the best way to go about that. And I did not. know the f64::hypot() method - rather nifty!
And good catch with the units, I missed that.

tgblackburn replied:
No worries, I will add this post-merge.
